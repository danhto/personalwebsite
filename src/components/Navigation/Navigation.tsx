import React from 'react'
import {MDBContainer,MDBNavbar, MDBNavbarBrand, MDBNavbarNav, MDBNavItem, MDBNavLink, MDBNavbarToggler, MDBCollapse, MDBFormInline} from 'mdbreact'

class Navigation extends React.Component {
    state = {
        isOpen: false
    }

    toggleCollapse = () => {
        this.setState ({
            isOpen: !this.state.isOpen
        })
    }

    render() {
        const { isOpen } = this.state

        return (
            <MDBContainer fluid>
                <MDBNavbar color="indigo" dark expand="md">
                <MDBNavbarBrand>
                    <strong className="white-text">DAN TO</strong>
                </MDBNavbarBrand>
                <MDBNavbarToggler onClick={this.toggleCollapse} />
                <MDBCollapse id="navbarCollapse3" isOpen={isOpen} navbar>
                    <MDBNavbarNav left>
                        <MDBNavItem>
                            <MDBNavLink to="/personalwebsite">Home</MDBNavLink>
                        </MDBNavItem>
                        <MDBNavItem>
                            <MDBNavLink to="/personalwebsite/about">About Me</MDBNavLink>
                        </MDBNavItem>
                        <MDBNavItem>
                            <MDBNavLink to="/personalwebsite/links">Links</MDBNavLink>
                        </MDBNavItem>
                        <MDBNavItem>
                            <MDBNavLink to="/personalwebsite/apps">Apps</MDBNavLink>
                        </MDBNavItem>
                    </MDBNavbarNav>
                    <MDBNavbarNav right>
                        <MDBNavItem>
                            <MDBFormInline waves>
                            <div className="md-form my-0">
                                <input className="form-control mr-sm-2" type="text" placeholder="Search" aria-label="Search" />
                            </div>
                            </MDBFormInline>
                        </MDBNavItem>
                    </MDBNavbarNav>
                </MDBCollapse>
                </MDBNavbar>
            </MDBContainer>
        )
    }
}

export default Navigation