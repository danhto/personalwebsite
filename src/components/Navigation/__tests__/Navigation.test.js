import {configure, shallow} from 'enzyme';
import React from 'react';
import Navigation from '../Navigation';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('<Navigation /> component tests', () => {
    it('expect component to mount', () => {
        const about = shallow(<Navigation />);
        expect(about).toBeTruthy();
    });
    
    it('match snapshot', () => {
        const about = shallow(<Navigation />);
        expect(about).toMatchSnapshot();
    })
});