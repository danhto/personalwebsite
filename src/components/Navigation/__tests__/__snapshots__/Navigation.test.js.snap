// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> component tests match snapshot 1`] = `ShallowWrapper {}`;
